packages/@aws-cdk/aws-stepfunctions-tasks/lib/apigateway/base.ts

```diff
@@ -0,0 +1,69 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+import { AuthType, InvokeApiGatewayApiBaseProps } from './base-types';
+
+/**
+ * Base ApiGateway Invoke Task
+ * @internal
+ */
+export abstract class InvokeApiGatewayApiBase extends sfn.TaskStateBase {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  private readonly baseProps: InvokeApiGatewayApiBaseProps;
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  protected abstract readonly apiEndpoint: string;
+  protected abstract readonly arnForExecuteApi: string;
+  protected abstract readonly stageName?: string;
+
+  constructor(scope: Construct, id: string, props: InvokeApiGatewayApiBaseProps) {
+    super(scope, id, props);
+
+    this.baseProps = props;
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+    validatePatternSupported(this.integrationPattern, InvokeApiGatewayApiBase.SUPPORTED_INTEGRATION_PATTERNS);
+
+    if (this.integrationPattern === sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN) {
+      if (!sfn.FieldUtils.containsTaskToken(this.baseProps.headers)) {
+        throw new Error('Task Token is required in `headers` for WAIT_FOR_TASK_TOKEN pattern. Use JsonPath.taskToken to set the token.');
+      }
+    }
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask() {
+    return {
+      Resource: integrationResourceArn('apigateway', 'invoke', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        ApiEndpoint: this.apiEndpoint,
+        Method: this.baseProps.method,
+        Headers: this.baseProps.headers?.value,
+        Stage: this.stageName,
+        Path: this.baseProps.apiPath,
+        QueryParameters: this.baseProps.queryParameters?.value,
+        RequestBody: this.baseProps.requestBody?.value,
+        AuthType: this.baseProps.authType ? this.baseProps.authType : 'NO_AUTH',
+      }),
+    };
+  }
+
+  protected createPolicyStatements(): iam.PolicyStatement[] {
+    if (this.baseProps.authType === AuthType.NO_AUTH) {
+      return [];
+    }
+
+    return [
+      new iam.PolicyStatement({
+        resources: [this.arnForExecuteApi],
+        actions: ['ExecuteAPI:Invoke'],
```

Comment:
I have ['execute-api:Invoke'] for the actions which I believe I got from the aws documentation... but I'm also trying to figure out why it's not working. Trying this other value now instead

Reply:
@eikeon good catch, I think what you have is what the [documentation](https://docs.aws.amazon.com/step-functions/latest/dg/api-gateway-iam.html) recommends as well. i'm not sure if it's case sensitive, but let's fix this up as well @ayush987goyal ?

Reply:
Whoops missed this! Not sure why it was working with this too. Fixed now. Thanks!

Reply:
Was/is there an integration test for this case? -- not sure I saw one when I looked. I did end up getting one of my cases working with the as documented value. Still not sure why I can't get this policy statement working in the other -- at least related to this ticket a bit if anyone has ideas at a glance
```
            new ShellScriptAction({
                actionName: "ValidateBeta",
                additionalArtifacts: [postBuildArtifact],
                useOutputs: {
                    API_REGIONAL_DOMAIN_NAME: pipeline.stackOutput(
                        beta.domainRegionalDomainName
                    ),
                    API_NAME: pipeline.stackOutput(beta.domainName),
                },
                commands: ["aws sts get-caller-identity", "npm run validate"],
                rolePolicyStatements: [
                    new iam.PolicyStatement({
                        resources: ["arn:aws:execute-api:*:*:*"],
                        actions: ["execute-api:Invoke"],
                    }),
                ],
            })
```

Reply:
I see the integration test now.

Reply:
Figured out my ShellScriptAction bit... my script was trying to read from ENV variables. Now using SDK to read them so it picks them up from wherever they are in the CodeBuild case. And the library I was using for signature-v4 didn't care the least when the credentials were undefined :(